nion/swift/model/Utility.py

```diff
@@ -432,6 +433,50 @@ def do_sample() -> None:
     threading.Thread(target=do_sample).start()
 
 
+def simplify_filename(filename: str, replacement_char: str = '_', maximum_length: int = 64) -> str:
+    # This function replaces any illegal characters in a file name. Not a full path.
+    # macOS illegal characters = \0 /
+    # linux illegal characters = \0 /
+    # Windows illegal characters = < > : / \ | ? * " ASCII values 0-31 (non-printable chars)
+    # Windows illegal filenames = CON PRN AUX NUL COM1 COM2 COM3 COM4 COM5 COM6 COM7 COM8 COM9
+    #                             LPT1 LPT2 LPT3 LPT4 LPT5 LPT6 LPT7 LPT8 LPT9
+    # those files names are illegal with or without a suffix, upper and lower case
+    # Windows file names cannot start or end with periods or spaces
+
+    assert maximum_length > 5
+    assert len(replacement_char) == 1
+
+    # since windows is the most restrictive and covers the others, just focus on that
```

Comment:
I'm not sure whether I like this idea -- but we could make the restrictions platform specific. The downside is one could save a quirky file on macOS and not be able to load it on Windows - but that may be preferable to strange limitations on macOS for macOS-only users. Any thoughts?